File: 백준/Silver/1920. 수 찾기/수 찾기.java

import java.util.Arrays;
import java.util.Scanner;

public class Main {

	public static void main(String[] args) {

		Scanner sc = new Scanner(System.in);

		int n = sc.nextInt();
		int[] arr = new int[n];

		for (int i = 0; i < n; i++) {
			arr[i] = sc.nextInt();
		}
		Arrays.sort(arr);

		StringBuilder sb = new StringBuilder();
		int cnt = sc.nextInt();
		for (int i = 0; i < cnt; i++) {
			int value = sc.nextInt();
			if (Arrays.binarySearch(arr, value) < 0) {
				sb.append(0+"\n");
			} else {
				sb.append(1+"\n");
			}
		}
		System.out.println(sb);

	}

}

File: 백준/Silver/1920. 수 찾기/수 찾기.py
'''
[구상]
    1. 오름차순 정렬
    2. 이진 탐색
'''


def binary_search(target):
    start = 0
    end = n - 1

    while start <= end:
        middle = (start + end) // 2
        if (arr[middle] == target):
            return 1
        elif (arr[middle] > target):
            end = middle - 1
        else:
            start = middle + 1
    return 0


n = int(input())
arr = list(map(int, input().split()))

m = int(input())
find = list(map(int, input().split()))

arr.sort()

for target in find:
    print(binary_search(target))
